engine/src/main/java/io/camunda/zeebe/engine/api/records/RecordBatch.java

```diff
@@ -98,4 +98,10 @@ public void forEach(final Consumer<? super ImmutableRecordBatchEntry> action) {
   public Spliterator<ImmutableRecordBatchEntry> spliterator() {
     return recordBatchEntries.spliterator();
   }
+
+  public static class ExceededBatchRecordSizeException extends RuntimeException {
+    public ExceededBatchRecordSizeException(final String message) {
+      super(message);
+    }
```

Comment:
🔧 The message is unused but it is filled with useful data. Let's move this data into the exception and use it to enhance the rejection message.